students/johnpharmd/lesson09/mailroom_OO.py

```diff
@@ -0,0 +1,120 @@
+        super().__init__(Donor.title, Donor.donor, Donor.donation)
+
+    @property
+    def donorgroup(self):
+        """getter for self.donors"""
+        return [donor.donor for donor in self.donors]
+
+    @donorgroup.setter
+    def donorgroup_new_donor(self, new_donor):
+        """enables addition of new donor"""
+        self.donors.append(new_donor)
+
+    def withdraw(self, title, last_name):
+        """given donor last name as string, removes donor from self.donors"""
+        for d in self.donors[:]:
+            if last_name in d.donor and d.donor[last_name]['title'] == title:
+                self.donors.remove(d)
+
+    def get_list(self):
+        self.donor_list = []
+        for d in self.donors:
+            for last_name in d.donor:
+                self.donor_list.append([d.donor[last_name]['donations'],
+                                        last_name,
+                                        d.donor[last_name]['num_donations']])
+        return sorted(self.donor_list, reverse=True)
+
+    def get_report(self):
+        print()
+        psv = ['Donor Name', '| Total Given', '| Num Gifts',
+               '| Average Gift']
+        print('{:<15}{:>12}{:>12}{:>12}'.format(psv[0], psv[1],
+              psv[2], psv[3]))
+        for i in range(55):
+            print('-', end='')
+        print()
+        new_list = self.get_list()
+        for donor_list in new_list:
+            formatted_donor = ('{:<15}'.format(donor_list[1])
+                               + '{}{:>10}'.format(' $', donor_list[0])
+                               + '{:>13}'.format(donor_list[2])
+                               + '{}{:>11}'.format(' $',
+                               donor_list[0] // donor_list[2]))
+            # if __name__ == '__main__':
+            #     print(formatted_donor)
+
+    def save_data(self):
+        with open('test.txt', 'w') as outfile:
+            outfile.write('This is current DonorGroup data:\n' +
+                          str(self.donorgroup))
+
+
+class UI(DonorGroup):
+    def __init__(self):
+        self.donors = (DonorGroup.donorgroup)
+        self.menu_dict = {'1': DonorGroup.get_list,
+                          '2': DonorGroup.get_report, 'q': sys.exit}
+        self.main_text = '\n'.join((
+                                    'Choose from the following:',
+                                    '"1" - Get a List,',
+                                    '"2" - Create a Report, or',
+                                    '"q" to Quit: '
+                                  ))
+        super().__init__()
+        while True:
+            print('\nMain Menu:')
+            response = input(self.main_text)
+            print()
+            try:
+                if response == 'q':
+                    print('Program execution completed.')
+                self.menu_dict[response](self)
+            except KeyError:
+                print('\nThat selection is invalid. Please try again.')
```

Comment:
the main program CLI does not need to be a class, you just need to represent a single donor and a collection of donors as objects.

for your menu switch, you could do something like this:
```
donor_group = DonorGroup()  # pass whatever is required into constructor
def send_thank_you():
    # your function to handle user input and interact with your class objects
menu_dict = {"1": thank_you, "2": create_report}
```